### PR TITLE
Update ValidationResult interface

### PR DESCRIPTION
**Pull request by Nasdan**

### CODE REVIEW

File: src/model/result.model.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 export interface ValidationResult {
   type: string;
   succeeded: boolean;
-  message: string;
+  message: string | string[];
```
Comment (brauliodiez) [inline src/model/result.model.ts]: Validation result is a single message, when we allow to create multiple messages.i the validation means that we can offer several messages (e.g. a validator that tells you that time cannot be past or you need to book five days I'm advance) but always the result will return only one

Comment (Nasdan) [reply src/model/result.model.ts]: So, I don't know where we are using the message as `string[]` provided by FieldValidationArgs or RecordValidationArgs

Comment (brauliodiez) [reply src/model/result.model.ts]: Let's say we want to create an IBAN validator this validator conjtains

ES7921000813610123456789
(this contains encoded country name, bank code, cc code etc...)
And you don't want to limit the valdiation just to "Invalid IBAN", you want to provide meaningfull messages:

- Invalid country code
- Invalid dc number
- Invalid bank number
- Invalid length

You just want to return a single error message in the validator but with a rich meaninful.

**Comment by Nasdan** [comment]
I've restore the previous code. I've only remove the uneeded internal types in `d.ts` file

**Pull request merged**